README.md

```diff
@@ -422,6 +432,55 @@ When storing logs on disk, we recommend using Logrotate in truncate mode.
 
 You can use [jq](https://stedolan.github.io/jq/) to traverse JSON log files, when you're not running Kiev in *development mode*.
 
+### Suffixing `tree_path`
+
+Most of the time, Kiev handles tracing just fine by managing `tree_path` all by itself. The trace can endure 13 synchronous requests and/or background jobs on every level, in any combination.
+
+However, there can be exceptions.
+
+For example, when sending a message to an MQ topic (e. g. in RabbitMQ or SNS) there no way to send distinct messages to individual subscribers. And since all subscribers receive *identical* messages, with no intervention they will log identical `tree_path`s. Then, if they do loggable actions inside (perform other requests or add jobs) you will have no way of knowing which of all subscribers did this. This is precisely what `tree_path` was designed to solve, but in this case it fails. However, it can be fixed.
+
+To keep the request trace clear and unambiguous
+
+* send out the message with data from `Kiev::SubrequestHelper.payload`
+* pick a unique suffix for each subscriber of group thereof that you wish to identify, then append it to the `tree_path` when receiving the message
+  * Don't waste memory and keep the suffix short, preferably a single character. And if your subscribers can be sensibly ordered, having corresponding lexicographic ordering of their suffixes might help too. Whether or not to use letters (that Kiev already uses) is up to you.
+
+Kiev provides a helper method for the subscriber's side in [Shoryuken](https://github.com/phstc/shoryuken), where this problem can arise with messages from subscriptions to SNS:
+
+```ruby
+# Suffix a single worker class:
+class MyWorker
+  include Shoryuken::Worker
+  Kiev::Shoryuken.suffix_tree_path(self, "X")
+  # ...
+end
+
+# Or use a suffix process-wide:
+Shoryuken.configure_server do |config|
+  Kiev::Shoryuken.suffix_tree_path(config, "X")
+end
+```
+
+Here's an example of the possble `tree_path` sequence you could get by configuring two consumers with suffixes `1` and `2` (note ordering by `tree_path`):
+
+| `tree_path`  | Meaning |
```

Comment:
use https://www.tablesgenerator.com/markdown_tables to make table more readable in text mode

Reply:
Right. Will do.
I'm curious if this kind of information would be represented better with nested lists, like

* `A` Something
  * `AB` Something
    * `ABA` Something
    * `ABD` Something
  * `AD` Something
  * `AE` Something

Reply:
not sure - need to see example